tests/h5test.cpp

```diff
@@ -438,10 +445,13 @@ int main(int argc, char **argv) {
                      gv_3d_name[igv], component_name(c3d[ic]), use_real ? "_r" : "");
             master_printf("Checking %s...\n", name);
             if (!check_3d(funky_eps_3d, a, splitting, Sf3[iS], Ez, c3d[ic], gv_3d[igv], use_real,
-                          gv_3d_rank[igv], name))
+                          gv_3d_rank[igv], name, temp_dir))
               return 1;
           }
       }
 #endif /* HAVE_HDF5 */
+
+  if (am_master()) rmdir(temp_dir);
```

Comment:
This won't work.  `rmdir` only works for empty directories.

See also [this stackoverflow post](https://stackoverflow.com/questions/2256945/removing-a-non-empty-directory-programmatically-in-c-or-c).     It seems like we could implement the POSIX `ntfw` solution pretty easily to add a `meep::delete_directory` function.

Reply:
i.e. add
```cxx
#include <ftw.h>
static int rmpath(const char *path, const struct stat *s, int type, struct FTW *ftw) {
    (void) s; (void) ftw;
    return remove(path) < 0 ? -1 : 0;
}

// equivalent to rm -rf path
void delete_directory(const char *path) {
    nftw(argv[1], rmFiles,10, FTW_DEPTH|FTW_MOUNT|FTW_PHYS);
}
```

Reply:
Done.